### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by tumainimaganiko**
In this project I did the following :

- Add Redux Toolkit `(npm install react-redux @reduxjs/toolkit)`.
- Create a directory that contains all my Redux logic `(/src/redux)`.
- Configure a Redux store `(/src/redux/store.js)`.
- Define a `slice` of state for `books` that:
   - Includes an **array of books** (initial state: empty array)
   - Includes a **reducer that adds a book**
   - Includes a **reducer that removes a book**
- Define a `slice` of state for `categories` that:
   - Includes an **array of categories** (initial state: empty array)
   - Includes a **reducer that checks the status** and always returns "Under construction" .
- Structure my application files using a `Redux folder`

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const initialstate = {
+  books: [],
+};
+
+export const booksSlice = createSlice({
+  name: 'books',
+  initialstate,
+  reducers: {
+    addBook: (state, action) => {
+      state.books = action.payload;
```
Comment (omarsalem7) [inline src/redux/books/booksSlice.js]: - [ ] Kindly consider fixing add the book to make it works properly by using the following snippet `state.books.push(action.payload);`

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @tumainimaganiko 😊, 

Good job so far! 👏
You have done well reaching this milestone point, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights

- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (tumainimaganiko) [reply src/redux/books/booksSlice.js]: Thank's for noticing that

**Review by divinecharlotte — APPROVED** [review]
# Hello @tumainimaganiko      😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say  👍🏻 ✅ 
### To Highlight 💯

- Good implementation of the requested changes👍 
- Descriptive pr 👍 
- Descriptive PR 👍 
 




## Congratulations! 🎉 



Cheers, and Happy coding!👏👏👏


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**